Shared/Coordinator/MainCoordinator.swift

```diff
@@ -147,10 +160,19 @@ class MainCoordinator: Coordinator<Void> {
 }
 
 #if IOS
+
+// MARK: - UserNotificationManagerDelegate
+
 extension MainCoordinator: UserNotificationManagerDelegate {
 
     nonisolated func userNotificationManager(willHandle deeplink: DeepLinkable) {
-        Task.onMainActor {
+        Task.onMainActorAsync {
+            // Cancelling other deeplink calls.
+            self.launchAndDeepLinkTask?.cancel()
+
+            // Wait until the launch is finished.
+            await self.launchAndDeepLinkTask?.value
+
             self.handle(deeplink: deeplink)
```

Comment:
Now we'll only handle the deep link once we're finished. Cancelling ensures that we don't attempt to deep link twice.